src/apps/recentlyUsed.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Gets the list of apps that have been used recently by the authenticated user.
+ *
+ * @module apps/recentlyUsed
+ */
+
+import { validateInterval, validateLimit } from "../util";
+import * as config from "../configuration";
+import constants from "../constants";
+import logger from "../logging";
+
+// All apps returned by this query are DE apps, so the system ID can be constant.
+const recentlyUsedAppsQuery = `
+    SELECT DISTINCT
+        a.id,
+        'de' AS system_id,
+        a.name,
+        a.description,
+        a.wiki_url,
+        a.integration_date,
+        a.edited_date,
+        u.username,
+        EXISTS (
+            SELECT * FROM users authenticated_user
+            JOIN workspace w ON authenticated_user.id = w.user_id
+            JOIN app_category_group acg ON w.root_category_id = acg.parent_category_id
+            JOIN app_category_app aca ON acg.child_category_id = aca.app_category_id
+            WHERE authenticated_user.username = $1
+            AND acg.child_index = $2
+            AND aca.app_id = a.id
+        ) AS is_favorite,
+        TRUE AS is_public,
+        max(j.start_date) AS most_recent_start_date
+    FROM jobs j
+    JOIN users ju ON j.user_id = ju.id
+    JOIN apps a ON CAST(a.id AS text) = j.app_id
+    LEFT JOIN integration_data d ON a.integration_data_id = d.id
+    LEFT JOIN users u ON d.user_id = u.id
+    WHERE ju.username = $1
```

Comment:
Small thing here as well, but with the updated thing, we might benefit from an index on `(username, start_date)`. That'd remove the second-largest thing at ~4ms, the bitmap heap scan that's done after using the username index.

Reply:
I experimented with this a little too. The query planner did use the index in this case, but the difference wasn't really significant. [Without the index](https://explain.depesz.com/s/SKu6), my test query took roughly 22ms. [With the index](https://explain.depesz.com/s/WSrZ), it took roughly 21.2ms. After altering the query to have to deal with more rows, I saw a savings of about 20ms (out of roughly 500ms). That's a pretty small portion of the execution time. I think I'll leave the index out unless we find some cases where having it would help later.